### PR TITLE
Allow single element padding_values to be broadcasted to a structure for tf.data.Dataset.padded_batch

### PR DESCRIPTION
**Pull request by yongtang**
This PR is a follow up based on comment in https://github.com/tensorflow/tensorflow/issues/35900#issuecomment-576969545

In #35900, the issue was raised where if padding_values is a single elment
and the dataset has a structured shape, then tf.data.Dataset.padded_batch
will return and error.

This PR adds the support to "broadcast" padding_values to match the same
structure as the dataset, which could be convenient in many cases (avoid
repeat the padding_values into the stucture manually).

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by jsimsa — CHANGES_REQUESTED** [review]
Thanks Yong. Please update the documentation of the `padding_values` argument of `padded_batch` accordingly` and ideally also add an example to the docstring.

**Comment by yongtang** [comment]
Thanks @jsimsa. The PR has been updated with docstring expanded with a new code example. Please take a look and let me know if there are any issues.

File: tensorflow/python/data/ops/dataset_ops.py [hunk 1]
```diff
@@ -1421,6 +1421,14 @@ def padded_batch(self,
     * If the dimension is unknown, the component will be padded out to the
       maximum length of all elements in that dimension.
 
+    The `padding_values` argument determines the values used to pad each
+    component to the respective shape. The `padding_values` should have the
+    same structure as the input dataset. If `padding_values` is a single
+    element and the input dataset has multiple components, then the same
+    `padding_values` will be used to pad every component of the dataset.
+    If `padding_values` is a scalar, then its value will be broadcasted
+    to match the shape of each component.
```
Comment (jsimsa) [inline tensorflow/python/data/ops/dataset_ops.py:1430]: This should be included in the "Args" section of the docstring instead.

Comment (yongtang) [reply tensorflow/python/data/ops/dataset_ops.py:1430]: @jsimsa Thanks. Updated.

**Comment by yongtang** [comment]
@jsimsa Thanks for the review. The PR has been updated with docstring adjusted to args part. Please take a look.

**Review by jsimsa — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Pull request merged**